astropy/units/quantity_helper/function_helpers.py

```diff
@@ -187,7 +187,7 @@ def __call__(self, f=None, helps=None, module=np):
         np.fft.fft2, np.fft.ifft2, np.fft.rfft2, np.fft.irfft2,
         np.fft.fftn, np.fft.ifftn, np.fft.rfftn, np.fft.irfftn,
         np.fft.hfft, np.fft.ihfft,
-        np.linalg.eigvals, np.linalg.eigvalsh,
+        np.linalg.eigvals, np.linalg.eigvalsh, np.nanmedian,
```

Comment:
I thought it was really nice that this works so easily, but unfortunately, the override does not explicitly look for an `out` argument and hence I fear things may fail if one is present. Could you add a test? (be sure to check that that the right instance is returned, see other tests in `test_quantity_non_ufuncs`). My guess is that you'll need a separate routine that takes care of `out`. (Note that `nanmedian` can also get us into `Quantity.__array_ufunc__` if out is  a Quantity but the input is not, so one needs to turn out into a regular ndarray before passing on)

Reply:
Dang, I knew this was too simple to work properly, good catch! So is there any alternative to just adding another function to `function_helpers.py`?

Reply:
So, yes, just add a new function! Maybe the `dot_like` function can give inspiration.